src/plugins/explore/public/components/visualizations/table/cell_value.tsx

```diff
@@ -37,5 +72,84 @@ export const CellValue = (props: Props) => {
     });
   }, [setCellProps, value, textAlign, color, colorMode]);
 
-  return value;
+  const resolveUrl = (url: string) => {
+    return url.replace('${__value.text}', encodeURIComponent(value || ''));
+  };
+
+  const getAnchorPosition = () => {
+    if (!buttonRef.current || !columnWidth) return 'downCenter';
+    const textWidth = buttonRef.current.scrollWidth;
+    if (textWidth > columnWidth || columnWidth > 200) {
+      return textAlign === 'right' ? 'downRight' : 'downLeft';
+    }
+    return 'downCenter';
+  };
+
+  const applicableLinks =
+    dataLinks?.filter((link) => columnId && link.fields.includes(columnId)) || [];
+
+  if (!applicableLinks || applicableLinks.length === 0) {
+    return <span>{value}</span>;
+  }
+
+  if (applicableLinks.length === 1) {
+    const link = applicableLinks[0];
+    const resolvedUrl = resolveUrl(link.url);
+    return (
+      <a
+        href={resolvedUrl}
+        target={link.openInNewTab ? '_blank' : '_self'}
+        rel={link.openInNewTab ? 'noopener noreferrer' : undefined}
+      >
+        {value}
+      </a>
+    );
+  }
+
+  const contextMenuItems = applicableLinks.map((link) => ({
+    name: link.title,
+    href: resolveUrl(link.url),
+    target: link.openInNewTab ? '_blank' : '_self',
+    rel: link.openInNewTab ? 'noopener noreferrer' : undefined,
+  }));
+
+  if (setPopoverOpen && isPopoverOpen !== undefined) {
+    return (
+      <EuiPopover
+        button={
+          <button
+            ref={buttonRef}
+            type="button"
+            onClick={() => setPopoverOpen(true)}
+            style={{
+              cursor: 'pointer',
+              textDecoration: 'underline',
+              ...(columnWidth ? { maxWidth: `${columnWidth}px` } : {}),
+            }}
+          >
+            {value}
+          </button>
+        }
+        isOpen={isPopoverOpen}
+        closePopover={() => setPopoverOpen(false)}
+        panelPaddingSize="none"
+        anchorPosition={getAnchorPosition()}
+        repositionOnScroll
+      >
+        <EuiContextMenu
+          initialPanelId={0}
+          panels={[
+            {
+              id: 0,
+              title: i18n.translate('explore.stylePanel.table.cellValue.selectLink', {
+                defaultMessage: 'Select a link',
```

Comment:
What about not showing the title? And instead add a OUI `link` icon to the context menu items? https://oui.opensearch.org/1.21/#/display/icons